app/common/data/interfaces/user.py

```diff
@@ -167,3 +168,33 @@ def remove_all_roles_from_user(user: User) -> None:
     db.session.execute(statement)
     db.session.flush()
     db.session.expire(user)
+
+
+def create_invitation(
+    email: str,
+    organisation_id: uuid.UUID | None = None,
+    grant_id: uuid.UUID | None = None,
+    role: RoleEnum | None = None,
+) -> Invitation:
```

Comment:
Do we think we'll have the actual organisation/grant objects here? If so, can we pass those in? By using the full objects we have a better chance of typechecking catching us getting the order wrong. We could also switch these to kwarg-only for extra protection.